### ISSUE TITLE
[Roadmap] Extend MES Functionality

### ISSUE THREAD

**Issue by SchrodingersGat** [inventree/InvenTree#4826]
There are some disparate PRs in the works which tend towards more fully-featured MES functionality. Collecting them here for further discussion. I'd like to have a well-considered plan towards MES functionality:

### References

- https://github.com/inventree/InvenTree/pull/4824
- https://github.com/inventree/InvenTree/issues/2860
- https://github.com/inventree/InvenTree/issues/2817
- https://github.com/inventree/InvenTree/pull/4855
- https://github.com/inventree/InvenTree/issues/4860
- https://github.com/inventree/InvenTree/issues/4818

**Comment by matmair** [comment]
@SchrodingersGat With the roadmap label I can assume you support us moving into a MES direction?

**Comment by SchrodingersGat** [comment]
Yes, I'm scope creeping! My company and a number of others I know of already use inventree for MES practices. Mostly through build orders and plugins.

Some ideas of what we could add in:

- communication with shop floor machines
- build process instructions/ steps which integrates with commission tests
- improved tracking of complex build processes

Any other ideas?

Keep in mind that these additions should be "optional" for users - the current simplifie workflow should not be forced to change significantly.

**Comment by matmair** [comment]
I already have (bad) plugins for instructions and communications; Both are very much possible - a good UI for them is the real issue.

**Comment by wolflu05** [comment]
Still in work

**Comment by SchrodingersGat** [comment]
No, not stale

**Comment by SchrodingersGat** [comment]
Not stale

**Comment by SergeoLacruz** [comment]
Some thoughts from my side. We use Build Orders to control HWbuilds, mainly PCB assembly. I wrote something that allows to add additional data to the BO

https://github.com/SergeoLacruz/build_plugin

Using this and the report system we can create documents like request for quotation more or less automatically based on data in the system. Just an idea.
An additional thing I am thinking about: the change part request. I happens quite often that a BOM code needs to be changes in the very last minute based on engineering request or material shortage. Or the designer wants to build 20 boards but 5 with one component changed, just to try something.  the function in Inventree is substitutes. But substitutes influence the BOM not the build. The BOM shall not be changed. This is advanced. :-) Actually we just write a change document and add it to the attachments. OK for now. 
Finally, uploading a BOM does not allow to update. Just add or delete all. In latter case all allocations are gone. Some day soon we will write plugin for BOM import to fix this. 
Michael

**Comment by matmair** [comment]
This is beeing worked on in plugins. Not sure if this is still a goal for core @SchrodingersGat?

**Comment by SchrodingersGat** [comment]
Still need to work out the framework for this

**Comment by CLOUDCUMPUTING** [comment]
![Image](https://github.com/user-attachments/assets/a3dca08e-c407-492d-9413-a69fb13e05b5)
![Image](https://github.com/user-attachments/assets/ade5a586-f1ab-48a5-a8e0-7ca4a7537d15)
![Image](https://github.com/user-attachments/assets/25bccb97-e7e6-4d90-8e19-414be03e9d87)

**Comment by matmair** [comment]
@CLOUDCUMPUTING can you please explain this post? As it stands I marked this as span

**Comment by CLOUDCUMPUTING** [comment]
My broken jaw, in the period of abuse I'm still struggling with. Corporate governance manslaughter cover up for JEETANDRA and a serious spiteful high profile syndicate of serial killers 

Killing kittens in Red Magazine November 2020 and the next article are linked 
As is grenfell Duncan Lewis George Street Croydon the drum and monkey and the windmill arson INCITEMENT thru the YMCA EVOLVE HOUSING PARTNERSHIP TO GRENFELL 3 WEEKS BEFORE THE FIRE 

PLEASE HELP ME BECAUSE I AM A VULNERABLE PERSON WHO IS NOT BEING GIVEN HUMAN RIGHTS AND COMPENSATION APPEARS TO BE INTERCEPTED AND PAYING FOR A COVER OP CAMPAIGN BY DIGITAL GOV CROYDON AND THE OPEN HOUSE MAGAZINE IS HELPING THEM RECRUIT THE WHOLE TOWN AND SURROUNDING AREAS

**Comment by matmair** [comment]
@CLOUDCUMPUTING I feel for you if you are hurting but this is not a fitting forum, please refrain from posting about non-technical topics on our pages.
